frontend/app/src/ui/mod.rs

```diff
@@ -339,6 +359,40 @@ impl UI {
             .end((instant::Instant::now() - self.start_time).as_millis() as f64);
     }
 
+    fn on_key_set_match<F>(&mut self, set: Vec<&str>, callback: F)
```

Comment:
I didn't want to do a huge refactor, but above you're running a similar pattern a lot of times.
- Check for keys down
- Check if those same keys are not being ignored
- If true, do some action
- Add those keys to the ignored list

This function bundles up some of the boilerplate. I'm just using it for the new key combos, but it could be used for the existing ones as well.